sale_report_delivered/reports/sale_report.py

```diff
@@ -90,10 +91,15 @@ def _select(self):
             sum(signed_qty * unsigned_product_uom_qty) AS product_uom_qty,
             sum(signed_qty * unsigned_price_subtotal) AS price_subtotal,
             sum(COALESCE(-sub.amount_cost, signed_qty *
-                sub.unsigned_purchase_price * unsigned_product_uom_qty)) AS amount_cost,
+                ROUND(sub.unsigned_purchase_price * unsigned_product_uom_qty,
+                      {decimal_places}))) AS amount_cost,
             sum(signed_qty * unsigned_price_subtotal - COALESCE(-sub.amount_cost, signed_qty *
-                sub.unsigned_purchase_price * unsigned_product_uom_qty)) AS margin
-        """
+                ROUND(sub.unsigned_purchase_price * unsigned_product_uom_qty,
+                      {decimal_places}))) AS margin,
+            0 AS margin_percent
+        """.format(
+            decimal_places=self.env.company.currency_id.decimal_places
```

Comment:
This should use the currency of the sales order, not the company currency.

Reply:
I was about to put a fixed 2...
This is a way to put an appropriate value without complicating the query and penalizing performance.
When someone opens an issue I promise to contemplate it.

Reply:
Carlos, the query is not much complicated. Just JOIN res_currency and the decimal places are there. This is not correct and may lead to weird data that is very difficult to debug. Please change it.

Reply:
I don't know if you're not understanding the context or if you're too maniacal :stuck_out_tongue_winking_eye: , but I don't want to penalize even a single millisecond for a query that can already be quite heavy.
![Selección_524](https://user-images.githubusercontent.com/7701001/157482573-19834f26-a83f-4f08-9c55-992b6c11d506.png)

If you are afraid of "weird data that is very difficult to debug", approve this PR because it is precisely what it is solving. :smile:

Reply:
Changes done.
Making the changes costs me less than writing the FIXME and that we are always with the debate :sweat_smile: